src/App.js

```diff
@@ -1,26 +1,110 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { Component } from 'react';
 import './App.css';
+import contacts from './contacts.json';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+//set the state to 5 contacts (create the contactlist and set it to state)
+class App extends Component { 
+  state = {
+    contactList: contacts.slice(0, 5),
+  };
+
+// --- add the random contacts-------
+  addRandom = () => {
+    const copyList = [...this.state.contactList];
+    const random = contacts[Math.floor(Math.random() * contacts.length)];
+    if (!copyList.includes(random)) {
+      copyList.push(random);
+    }
+    this.setState({
+      contactList: copyList,
+    });
+  };
+
+  //-----Sort the contacts by name
+  sortByName = () => {
+    const copyList = [...this.state.contactList];
+    copyList.sort(function (a, b) {
+      if (a.name < b.name) {
+        return -1;
+      }
+      if (a.name > b.name) {
+        return 1;
+      }
+      return 0;
+    });
+
+    this.setState({
+      contactList: copyList,
+    });
+  };
+
+
+
+    //-----Sort the contacts by poulatirity
+  sortByPopularity = () => {
+    const copyList = [...this.state.contactList];
+    copyList.sort(function (a, b) {
+      if (a.popularity > b.popularity) {
+        return -1;
+      }
+      if (a.popularity < b.popularity) {
+        return 1;
+      }
+      return 0;
+    });
+
+    this.setState({
+      contactList: copyList,
+    });
+  };
+
+
+    //-----delete the contacts
+  deleteContact = (index) => {
+    const removeContact = this.state.contactList.filter(
```

Comment:
Does the variable `removeContact` contain a removed contact? Could there be a more descriptive name?